apps/hash-frontend/src/pages/drafts.page/draft-entities.tsx

```diff
@@ -0,0 +1,269 @@
+  LastEditedTimeRanges,
+} from "./draft-entities/draft-entities-filters";
+import { DraftEntity } from "./draft-entity";
+
+const isDateWithinLastEditedTimeRange = (params: {
+  date: Date;
+  lastEditedTimeRange: LastEditedTimeRanges;
+}) => {
+  const { date, lastEditedTimeRange } = params;
+  const now = new Date();
+  switch (lastEditedTimeRange) {
+    case "anytime":
+      return true;
+    case "last-24-hours":
+      return date >= subHours(now, 1);
+    case "last-7-days":
+      return date >= subDays(now, 7);
+    case "last-30-days":
+      return date >= subDays(now, 30);
+    case "last-365-days":
+      return date >= subDays(now, 365);
+    default:
+      return true;
+  }
+};
+
+export type SortOrder = "created-at-asc" | "created-at-desc";
+
+export const DraftEntities: FunctionComponent<{ sortOrder: SortOrder }> = ({
+  sortOrder,
+}) => {
+  const [filterState, setFilterState] = useState<DraftEntityFilterState>();
+
+  const { draftEntities, draftEntitiesSubgraph } = useDraftEntities();
+
+  const [
+    previouslyFetchedDraftEntityHistoriesData,
+    setPreviouslyFetchedDraftEntityHistoriesData,
+  ] = useState<StructuralQueryEntitiesQuery>();
+
+  const { data: draftEntityHistoriesData } = useQuery<
+    StructuralQueryEntitiesQuery,
+    StructuralQueryEntitiesQueryVariables
+  >(structuralQueryEntitiesQuery, {
+    variables: {
+      query: {
+        filter: {
+          any:
+            draftEntities?.map((draftEntity) => ({
+              equal: [
+                { path: ["uuid"] },
+                {
+                  parameter: extractEntityUuidFromEntityId(
+                    draftEntity.metadata.recordId.entityId,
+                  ),
+                },
+              ],
+            })) ?? [],
+        },
+        includeDrafts: true,
+        temporalAxes: fullDecisionTimeAxis,
+        graphResolveDepths: zeroedGraphResolveDepths,
+      },
+      includePermissions: false,
+    },
+    skip: !draftEntities,
+    onCompleted: (data) => setPreviouslyFetchedDraftEntityHistoriesData(data),
+  });
+
+  const draftEntityHistoriesSubgraph = useMemo(
+    () =>
+      draftEntityHistoriesData || previouslyFetchedDraftEntityHistoriesData
+        ? mapGqlSubgraphFieldsFragmentToSubgraph<EntityRootType>(
+            (draftEntityHistoriesData ??
+              previouslyFetchedDraftEntityHistoriesData)!
+              .structuralQueryEntities.subgraph,
+          )
+        : undefined,
+    [draftEntityHistoriesData, previouslyFetchedDraftEntityHistoriesData],
+  );
+
+  const { users } = useUsers();
+
+  const draftEntitiesWithCreatedAtAndCreators = useMemo(() => {
+    if (!draftEntities || !draftEntityHistoriesSubgraph || !users) {
+      return undefined;
+    }
+
+    return draftEntities.map((entity) => {
+      const firstRevision = getFirstRevision(
+        draftEntityHistoriesSubgraph,
+        entity.metadata.recordId.entityId,
+      );
+
+      const creator = users.find(
```

Comment:
This is to help me find where I need to update when we have non-user creators
```suggestion
      // @todo update to account for machine users – H-1514
      const creator = users.find(
```